dotnet/src/Skills/Skills.Web/Bing/BingConnector.cs

```diff
@@ -32,15 +32,15 @@ public BingConnector(string apiKey, ILogger<BingConnector>? logger = null)
     }
 
     /// <inheritdoc/>
-    public async Task<IEnumerable<string>> SearchAsync(string query, int count = 1, int offset = 0, CancellationToken cancellationToken = default)
+    public async Task<IEnumerable<string>> SearchAsync(string query, string relatedSite = "", int count = 1, int offset = 0, CancellationToken cancellationToken = default)
     {
         if (count <= 0) { throw new ArgumentOutOfRangeException(nameof(count)); }
 
         if (count >= 50) { throw new ArgumentOutOfRangeException(nameof(count), $"{nameof(count)} value must be less than 50."); }
 
         if (offset < 0) { throw new ArgumentOutOfRangeException(nameof(offset)); }
 
-        Uri uri = new($"https://api.bing.microsoft.com/v7.0/search?q={Uri.EscapeDataString(query)}&count={count}&offset={offset}");
+        Uri uri = new($"https://api.bing.microsoft.com/v7.0/search?q={Uri.EscapeDataString(query)}{(!String.IsNullOrWhiteSpace(relatedSite) ? $"&site:{relatedSite}" : "")}&count={count}&offset={offset}");
```

Comment:
Have you tested this? As far as I know, you should not provide `site` as an extra query parameter, but embed it in the query itself. You should also not use the full url (e.g. http://www.contoso.com) but just `contoso.com`.

Reply:
1. You are correct! No need for `&`, `site:` should go directly in the query, I was testing without it.
2. Full url works fine too.

Reply:
Updated the PR with requested changes.